netusage@30yavash.com/files/netusage@30yavash.com/desklet.js

```diff
@@ -152,26 +152,18 @@ MyDesklet.prototype = {
         try {
            if (this._device != "null") {
                    if (this.useExtendedDisplay) {
-                         if(this.extendedDisplay == "classic" ) {
-
-                                GLib.spawn_command_line_sync('vnstati -s -ne -i ' + this._device + ' -o /tmp/vnstatlmapplet.png');
-                         }
-                         if(this.extendedDisplay == "classicPlus" ) {
-
-                               GLib.spawn_command_line_sync('vnstati -vs -ne -i ' + this._device + ' -o /tmp/vnstatlmapplet.png' );
-                         }
-
                          if(this.extendedDisplay == "userDefined" ) {
                                if (this.useVnstatiCommandString) {
                                      GLib.spawn_command_line_sync('vnstati ' + this.vnstatiCommandString + ' -ne -i ' + this._device + ' -o /tmp/vnstatlmapplet.png');
                                } else {
                                      GLib.spawn_command_line_sync('vnstati -s -ne -i ' + this._device + ' -o /tmp/vnstatlmapplet.png');
                                }
+                         } else {
+                                 GLib.spawn_command_line_sync('vnstati ' + this.extendedDisplay + ' -ne -i ' + this._device + ' -o /tmp/vnstatlmapplet.png');
```

Comment:
It might be better to try using `Util.spawnCommandLineAsync` or [spawn_async](https://github.com/linuxmint/Cinnamon/blob/bbbf728a01845a151f9da4d50a74804f2af53af7/js/misc/util.js#L91) here instead, so if the process misbehaves it doesn't block the main thread.

Reply:
@jaszhix I have a sense of deja vu on this as I originally suggested the same to Clem who wrote vnstat@cinnamon.com which this is based on. The problem there was that `vnstat` had to complete before the picture which `vnstati` generates can be displayed so it could sometimes be picking up an image which was days out of date and we eventually had to revert the change. 

Here the problem is similar in that I only use a single temp file for all the instances and if one does it asynchronously one picks up an  image from a random instance which is very confusing as they all keep changing! 

If you can suggest a simple way of defining a path to a different image for each instance I am very happy to try an async version but, as I am away over Christmas, I would prefer to leave it until the next version.

Reply:
One other thing I noticed was this might be causing a slow memory leak when `imageWidget` is re-parented. You might also want to add

```js
this.imageWidget.destroy_all_children();
```
before
```js
this.imageWidget.set_child(b);
```